File: src/#1161/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

var maxLevelSum = function (root) {
  let total = [];
  helper(root, 0);
  let maxLv = 0;
  let maxNum = Number.MIN_SAFE_INTEGER;

  total.forEach((v, i) => {
    if (v > maxNum) {
      maxLv = i;
      maxNum = v;
    }
  });
  return maxLv + 1;

  function helper(root, level) {
    if (root === null) {
      return;
    }
    if (level >= total.length) {
      total.push(root.val);
    } else {
      total[level] += root.val;
    }

    helper(root.left, level + 1);
    helper(root.right, level + 1);
  }
};

File: src/#1161/solution.go
package solution

import "math"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxLevelSum(root *TreeNode) int {
	total := []int{}
	helper(root, &total, 0)
	maxLv := -1
	maxNum := math.MinInt32
	for i, v := range total {
		if v > maxNum {
			maxLv = i
			maxNum = v
		}
	}
	return maxLv + 1
}

func helper(root *TreeNode, total *[]int, level int) {
	if root == nil {
		return
	}
	if level > len(*total) {
		*total = append(*total, root.Val)
	} else {
		(*total)[level] += root.Val
	}
	helper(root.Left, total, level+1)
	helper(root.Right, total, level+1)

}
